File: 2588-count-the-number-of-beautiful-subarrays/2588-count-the-number-of-beautiful-subarrays.cpp
class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums) {
        long long cnt = 0;
        int x = 0;
        unordered_map<int, int> mp;
        
        for(int i=0; i<nums.size(); i++) {
            x = x ^ nums[i];
            
            if(x == 0) cnt++;
            
            if(mp.find(x) != mp.end()) {
                cnt += mp[x];
            }
            mp[x]++;
        }
        
        return cnt;
    }
};

File: 2588-count-the-number-of-beautiful-subarrays/2588-count-the-number-of-beautiful-subarrays.java
class Solution {
    public long beautifulSubarrays(int[] nums) {
        long cnt = 0;
        HashMap<Integer, Integer> mp = new HashMap<>();
        int x = 0;
        
        for(int i=0; i<nums.length; i++) {
            x = x ^ nums[i];
            
            if(x == 0) cnt++;
            
            if(mp.containsKey(x)) {
                cnt += mp.get(x);
                mp.put(x, mp.get(x) + 1);
            } else {
                mp.put(x, 1);
            }
        }
        
        return cnt;
    }
}